extensions/sql-migration/src/dashboard/sqlServerDashboard.ts

```diff
@@ -0,0 +1,510 @@
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { MigrationLocalStorage } from '../models/migrationLocalStorage';
+import * as loc from '../models/strings';
+import { MigrationStatusDialog } from '../dialog/migrationStatusDialog';
+
+interface IActionMetadata {
+	title?: string,
+	description?: string,
+	link?: string,
+	iconPath?: { light: string | vscode.Uri; dark: string | vscode.Uri },
+	command?: string;
+}
+
+const maxWidth = 800;
+
+export class DashboardWidget {
+
+	private _migrationStatusCardsContainer!: azdata.FlexContainer;
+	private _view!: azdata.ModelView;
+	/**
+	 * Creates new instance of dashboard
+	 */
+	constructor(private _root: string) {
+	}
+
+	public register(): Promise<void> {
+		//Migrations.clearMigrations();
+		return new Promise<void>(resolve => {
+			azdata.ui.registerModelViewProvider('migration.dashboard', async (view) => {
+				this._view = view;
+				const container = view.modelBuilder.flexContainer().withLayout({
+					flexFlow: 'column',
+					width: '100%',
+					height: '100%'
+				}).component();
+				const header = this.createHeader(view);
+
+				const tasksContainer = await this.createTasks(view);
+
+				container.addItem(header, {
+					CSSStyles: {
+						'background-image': `url(${vscode.Uri.file(this.asAbsolutePath('images/background.svg'))})`,
+						'width': `1100px`,
+						'height': '300px',
+						'background-size': '100%',
+					}
+				});
+				header.addItem(tasksContainer, {
+					CSSStyles: {
+						'width': `${maxWidth}px`,
+						'height': '150px',
+					}
+				});
+
+				header.addItem(await this.createFooter(view), {
+					CSSStyles: {
+						'margin-top': '20px'
+					}
+				});
+
+				const mainContainer = view.modelBuilder.flexContainer()
+					.withLayout({
+						flexFlow: 'column',
+						width: '100%',
+						height: '100%',
+						position: 'absolute'
+					}).component();
+				mainContainer.addItem(container, {
+					CSSStyles: { 'padding-top': '25px', 'padding-left': '5px' }
+				});
+				await view.initializeModel(mainContainer);
+
+				this.refreshMigrations();
+				resolve();
+			});
+		});
+	}
+
+	private createHeader(view: azdata.ModelView): azdata.FlexContainer {
+		const header = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: maxWidth,
+		}).component();
+		const titleComponent = view.modelBuilder.text().withProperties({
+			value: loc.DASHBOARD_TITLE,
+			CSSStyles: {
+				'font-size': '36px',
+				'font-family': 'Segoe UI'
+			}
+		}).component();
+		const descComponent = view.modelBuilder.text().withProperties({
+			value: loc.DASHBOARD_DESCRIPTION,
+			CSSStyles: {
+				'font-size': '12px',
+			}
+		}).component();
+		header.addItems([titleComponent, descComponent], {
+			CSSStyles: {
+				'width': `${maxWidth}px`,
+				'padding-left': '20px'
+			}
+		});
+
+		return header;
+	}
+
+	private asAbsolutePath(filePath: string): string {
+		return path.join(this._root || '', filePath);
```

Comment:
If this can be undefined then the parameter should reflect that. If it can't then the || is unnecessary